pytorch_lightning/trainer/supporters.py

```diff
@@ -371,6 +370,64 @@ def __init__(self, loaders: Any, mode: str = 'min_size'):
         if self.mode == 'max_size_cycle':
             self._wrap_loaders_max_size_cycle()
 
+        self.loaders_iter_state_dict = None
+
+    def state_dict(self, num_batches_processed: int):
+        """
+        The state dict includes all states from wrapped dataloaders and their samplers through the
+        ``CaptureIterableDataset`` and fast-forward samplers.
+
+        Args:
+            num_batches_processed: The number of batches processed so far, needed because the individual dataloaders
+                may have already prefetched more batches by the time a state dict is requrested.
+        """
+        if not fault_tolerant_enabled():
+            return DataLoaderDict()
+
+        def state_dict_fn(dataloader: DataLoader, iterator: Iterator) -> Dict:
+            # find next worker if multiple workers were used
+            state = find_current_worker(iterator)
+            if isinstance(dataloader.dataset, CaptureIterableDataset):
+                # the sampler state dict are extracted in ``CombinedLoaderIterator``
+                if iterator is not None and getattr(iterator, "_sampler_state_dict", None) is not None:
+                    state.update(iterator._sampler_state_dict[0])
+            else:
+                # fetch directly from fast forward sampler
+                state.update(dataloader.fast_forward_sampler.state_dict(num_batches_processed))
+            return DataLoaderDict(state)
+
+        return apply_to_collections(self.loaders, self._iterator.loader_iters, (Iterator, DataLoader), state_dict_fn)
+
+    def load_state_dict(self, state_dict):
+        self.loaders_iter_state_dict = state_dict
+
+        def mock_reset_fn(self, *_, **__):
+            pass
+
+        # delay reset call, will be reset in on_restart
+        _MultiProcessingDataLoaderIter._ori_reset = _MultiProcessingDataLoaderIter._reset
```

Comment:
it's not obvious what ori means

```suggestion
        _MultiProcessingDataLoaderIter._original_reset = _MultiProcessingDataLoaderIter._reset
```